pkg/cluster/cluster.go

```diff
@@ -281,5 +287,110 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 		hiveClusterManager:                hiveClusterManager,
 		now:                               func() time.Time { return time.Now() },
 		openShiftClusterDocumentVersioner: new(openShiftClusterDocumentVersionerService),
-	}, nil
+	}
+
+	if doc.OpenShiftCluster.UsesWorkloadIdentity() {
```

Comment:
I don't think this whole block necessarily needs to be separated out into another function, since it's specific to the initialization of the cluster manager, which is what this function is for. If your concern is that it's long and distracting from the high-level idea, I think that will be addressed once I've pulled a few of the more complex pieces into helper functions based on some of your other suggestions.